model/src/main/kotlin/utils/SimplePackageConfigurationProvider.kt

```diff
@@ -33,9 +33,14 @@ import org.ossreviewtoolkit.model.readValue
  * Throws an exception if there is more than one configuration per [Identifier] and [Provenance].
  */
 class SimplePackageConfigurationProvider(
-    configurations: Collection<PackageConfiguration> = emptyList()
+    configurations: Collection<PackageConfiguration>
 ) : PackageConfigurationProvider {
     companion object {
+        /**
+         * A provider without any package configurations.
+         */
+        val EMPTY = SimplePackageConfigurationProvider(emptyList())
```

Comment:
Having this constant in `PackageConfigurationProvider` would make the calling code a bit nicer, but it would make the interface depend on the implementation. But I think that's okay, because both files are next to each other anyway.

Reply:
> but it would make the interface depend on the implementation.

That's a no-go for me, sorry. Would you then prefer dropping this change, as that's what @fviernau seems to prefer (see below)?

Reply:
You could also make the interface a functional interface and provide the empty object like this:
```
fun interface PackageConfigurationProvider {
    companion object {
        val EMPTY = PackageConfigurationProvider { _, _ -> null }
    }

    ...
}
```

Regarding the discussion below I agree that for a class that already has a no-argument constructor the EMPTY object is less important than otherwise.

Reply:
I must be missing something: Where is the no-argument constructor for `SimplePackageConfigurationProvider`? I only see a one-argument constructor with a default argument (in the original code), or is it that what you effectively mean by "no-argument"?

Reply:
> You could also make the interface a functional interface and provide the empty object like this:

Done.

Reply:
> I must be missing something: Where is the no-argument constructor for `SimplePackageConfigurationProvider`? I only see a one-argument constructor with a default argument (in the original code), or is it that what you effectively mean by "no-argument"?

Yes, that's what I meant. Although technically without `@JvmOverloads` there is not really a no-argument constructor :)

Reply:
Very interestingly, using

    val EMPTY = PackageConfigurationProvider { _, _ -> null }

or

    val EMPTY = object : PackageConfigurationProvider {
        override fun getPackageConfiguration(packageId: Identifier, provenance: Provenance): PackageConfiguration? {
            return null
        }
    }

makes `EvaluatorTest` fail, but using

    val EMPTY = SimplePackageConfigurationProvider(emptyList())

succeeds. I'm currently clueless why that is.

Reply:
This looks more or less like a Kotlin compiler bug to me. As I don't have the time to investigate this further, I've reverted to the `val EMPTY = SimplePackageConfigurationProvider(emptyList())` approach in the `SimplePackageConfigurationProvider` class.

Reply:
Maybe it's related to your compile error, over the last weeks I had to delete my Gradle build cache a few times, because of weird compilation errors that would not go away even with cleaning the project and the `--rerun-tasks` option.

Reply:
BTW, it's not a compile error, but a compile**r** error, in the sense that (bogus?) byte code might be generated, as the error occurs only at runtime, not at build time.